### PR TITLE
Exploration of simple flagging options

### PR DESCRIPTION
**Pull request by caldwellst**
Hi Monica! We can chat about this to discuss more, but put up some exploration I did. I was looking at various options that could be used to flag outliers in time series data, first by normalising the data to be daily for all countries then testing various flags. I have produced a summary graph if we were using a wide # of flags and when we would be flagging, but lots more we can do there. Let's chat about this. Sorry it's in R and not python, let me know if you want to restructure the system.

### CODE REVIEW

**Review by turnerm — APPROVED** [review]
Looks great! Do you have a sense if a particular flag or a combo might be close to what we're looking for? 

One small thing, maybe it was an oversight but I wouldn't check the data or figures into the repo. We could set up a place on Google Drive if you think syncing is needed.

File: analysis/r/pa-global-monitoring/exploration.R [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+# and then summing across days
+
+df_daily <- df %>%
+  mutate(
+    across(
+      contains("date"),
+      as.Date
+    ),
+    start_date = if_else(
+      displacement_start_date <= displacement_end_date,
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply analysis/r/pa-global-monitoring/exploration.R:43]: Yeah, there were a few instances where I was getting errors due to that. Not many, but was there!

File: analysis/r/pa-global-monitoring/utils.R [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  
+  quantile(x_check, probs = perc, na.rm = TRUE)
+}
+
+
+#' Get outliers from time series
+#' 
+#' Automatic detection using `tsoutliers()`.
+tsoutliers_clean <- function(x, row_numbers) {
+  outliers <- tsoutliers(x)
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply analysis/r/pa-global-monitoring/utils.R:30]: Not stupid at all! I was putting the `library(call)` in the `exploration.R` script even though it was in `utils.R`. Forgot to move it when I created the function in `utils.R` (which I had to do for the cases where no outliers were found).

File: analysis/r/pa-global-monitoring/exploration.R [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+      displacement_start_date,
+      displacement_end_date
+    ),
+    end_date = if_else(
+      displacement_start_date <= displacement_end_date,
+      displacement_end_date,
+      displacement_start_date
+    )
+  ) %>%
+  filter( # remove few cases where both start and end not available
```
[the remark this answers is not in the document]
Comment (caldwellst) [reply analysis/r/pa-global-monitoring/exploration.R:53]: Oooh smart, good catch!

File: analysis/r/pa-global-monitoring/utils.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (caldwellst) [reply analysis/r/pa-global-monitoring/utils.R:30]: And yeah, `tsoutliers()` comes from the `forecast` library, confusing there is a `tsoutliers` package itself!

[the remark this answers is not in the document]
Comment (caldwellst) [reply analysis/r/pa-global-monitoring/utils.R:30]: I didn't up using the flags from it in the final output, however. I think it's such more useful if you are looking for outlier flag on time series that are more "standard" and include seasonal and trend components, whereas I don't think displacement data really fits into that.

**Pull request merged**